### PR TITLE
Task/create meeting router

### PR DESCRIPTION
**Pull request by Matios102**

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
```
Comment (mateuszpolis) [inline .flake8:1]: if you installed something do pip freeze > requirements.txt

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,8 @@
 FROM python:3.9-slim
 WORKDIR /app
-COPY . /app
```
Comment (mateuszpolis) [inline Dockerfile:3]: why did you move that around

File: models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from .attendance import attendance
+from .classes import classes
```
Comment (mateuszpolis) [inline models/__init__.py]: put models back in one file as it was

File: models/meetings.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+from sqlalchemy.sql import func
+from database import metadata
+
+meetings = Table(
+    "meetings",
+    metadata,
+    Column("id", Integer, primary_key=True, index=True),
+    Column("teacher_id", Integer, ForeignKey("users.id")),
+    Column("class_id", Integer, ForeignKey("classes.id")),
+    Column("start_time", String),
```
Comment (mateuszpolis) [inline models/meetings.py]: remove start_time and end_time from here, its already in class

File: repositories/meeting_repository.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+from sqlalchemy.orm import Session
+from models import meetings
+from sqlalchemy import and_, select
+from models import students, students_classes
+
+
+def get_meeting_by_current_time(db: Session, current_time: datetime):
+    meeting_query = select(meetings).where(
+        and_(
+            meetings.c.start_time <= current_time.strftime("%H:%M"),
```
Comment (mateuszpolis) [inline repositories/meeting_repository.py]: get start_time and end_time from class

File: repositories/meeting_repository.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+def get_students_by_meeting_id(db: Session, meeting_id: int):
+    query = (
+        select(students)
+        .join(students_classes, students_classes.c.student_id == students.c.id)
+        .join(meetings, meetings.c.class_id == students_classes.c.class_id)
+        .filter(meetings.c.id == meeting_id)
+    )
+    return db.execute(query).fetchall()
+
+def cancel_meeting_by_id(db: Session, meeting_id: int):
```
Comment (mateuszpolis) [inline repositories/meeting_repository.py:79]: cancel does not mean delete. Add column canceled of type bool to the table and set that to true, then delete attendance for that meeting if there were any

File: routers/meetings_router.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+):
+    return fetch_meeting_by_current_time(db, current_user.id)
+
+@router.get("/meetings/{meeting_id}")
+async def get_meeting_by_id(
+    meeting_id: int, db: Session = Depends(get_db), current_user: UserPublic = Depends(get_current_user)
+):
+    return fetch_meeting_by_id(db, meeting_id, current_user.id)
+
+@router.get("/meetings/{meeting_id}/students", response_model=List[StudentBase])
```
Comment (mateuszpolis) [inline routers/meetings_router.py]: We don't want StudentBase here, we want to know if the student was present or not. Create a new schema that extends StudentBase with added field preset

File: routers/meetings_router.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+router = APIRouter()
+
+@router.get("/meetings/")
+async def get_meeting_by_current_time(
+    db: Session = Depends(get_db), current_user: UserPublic = Depends(get_current_user)
+):
+    return fetch_meeting_by_current_time(db, current_user.id)
+
+@router.get("/meetings/{meeting_id}")
```
Comment (mateuszpolis) [inline routers/meetings_router.py]: combine this meetings/{meeting_id} with meetings/{meeting_id}/students.
When we request the meeting we already want the list of students. 

You can look in classes how to join them

**Review by mateuszpolis — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 2 above)
Comment (HermanPlay) [reply Dockerfile:3]: It was me, when after this change you don't have to install requirements every time you build, the build will only copy new files and build the image and use cached requirements
